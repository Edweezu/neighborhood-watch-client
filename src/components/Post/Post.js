import React from 'react'
// import { findUser } from '../../helpers'
import MainContext from '../../contexts/MainContext';
import fileImage from '../../app-images/file-image-icon.png'
import { Link } from 'react-router-dom'
import config from '../../config'
import TokenService from '../../services/token-service'

class Post extends React.Component {

    static contextType = MainContext

    handleDeletePost = () => {

        //forgot you can just take postid from props in any function in the component. don't need to pass it into the function
        let { id } = this.props

        return fetch(`${config.API_ENDPOINT}/posts/${id}`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            }
        })
        .then(res => {
            if (!res.ok) {
                return res.json().then(e => Promise.reject(e))
            }
        })
        .then(resp => {
            this.context.deletePost(id)
            //reset state with updated posts, one less post
            window.location.reload()
        })
        .catch(err => {
            console.error(err)
        })
    }

    render () {
        // const { users=[] } = this.context
        const { id, subject, message, date_created, user } = this.props
        // const user = findUser(users, user_id) || []
        
        // console.log('users', users)
        // console.log('user id', user_id)
        // console.log('user', user)

        const nameCapitalized = user.username.charAt(0).toUpperCase() + user.username.slice(1)

        return (
            <section className='Post'>
                <div className='Post__userInfo'>
                    {nameCapitalized}, {user.city}
                </div>
                <div>
                    <h4><Link to={`/post-page/${id}`}>{subject}</Link></h4>
                    <p>{message}</p>
                    <figure>
                        <img src={fileImage} alt='default icon' width='100'/>
                        <figcaption>User uploaded image is going to replace this.</figcaption> 
                    </figure>
                    <div>
                        <p>Posted on {date_created}
                        </p>
                        <button type='button'>
                            Edit
                        </button>
                        <button type='button' onClick={this.handleDeletePost}>
                            Delete
                        </button>      
                    </div>
                    <div>
                        <button type='button'><i className="fas fa-thumbs-up"></i></button>
                    </div>
                </div>

            </section>
        )
    }
}

export default Post