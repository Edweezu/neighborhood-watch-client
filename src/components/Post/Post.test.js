import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Post from './Post'


describe('Post component', () => {
    const props = {
       id: 1,
       subject: 'random subject',
       message: 'random message',
       user: {
        username: 'random user'
       },
       image: 'random iamge text',
       number_of_comments: 1,
       user_logged_in: 1,
       date_created: "Wed Sep 25 2019 11:45:49 GMT-0700 (Pacific Daylight Time)",

    }

    it('renders the given component', () => {
        const wrapper = shallow(<Post />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<Post {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})