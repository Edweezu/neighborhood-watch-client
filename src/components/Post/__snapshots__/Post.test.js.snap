// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component renders the component's given props 1`] = `
<section
  className="Post"
>
  <div
    className="Post__userInfo"
  >
    Random user
    , 
  </div>
  <div>
    <h4>
      <Link
        to="/post-page/1"
      >
        random subject
      </Link>
    </h4>
    <p>
      random message
    </p>
    <figure>
      <img
        alt="default icon"
        src="random iamge text"
        width="100"
      />
      <figcaption>
        User uploaded image is going to replace this.
      </figcaption>
    </figure>
    <div>
      <p>
        Posted on Sep 25, 2019
      </p>
      <EditModal
        postid={1}
      />
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-thumbs-up"
        />
      </button>
      <span
        className="Post__commentContainer"
      >
        <i
          className="fas fa-comment"
        />
        <span
          className="Post__commentNumber"
        >
          1
        </span>
      </span>
    </div>
  </div>
</section>
`;

exports[`Post component renders the given component 1`] = `
<section
  className="Post"
>
  <div
    className="Post__userInfo"
  >
    , 
  </div>
  <div>
    <h4>
      <Link
        to="/post-page/undefined"
      />
    </h4>
    <p />
    <div>
      <p>
        Posted a few seconds ago
      </p>
      <EditModal />
    </div>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-thumbs-up"
        />
      </button>
      <span
        className="Post__commentContainer"
      >
        <i
          className="fas fa-comment"
        />
        <span
          className="Post__commentNumber"
        />
      </span>
    </div>
  </div>
</section>
`;
